src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -325,10 +373,90 @@ internal static bool ValidateParentTagName(string parentTag, ErrorSink errorSink
                     errorSink: errorSink);
         }
 
-        private static bool ValidateName(
-            string name,
-            bool targetingAttributes,
-            ErrorSink errorSink)
+        private static bool ValidateRequiredAttribute(string attribute, ErrorSink errorSink)
```

Comment:
Guess that since parsing validates the syntax as well, makes sense for parsing / validation to collect information used to instantiate a `TagHelperRequiredAttributeDescriptor` rather than the other way round. The one-pass parser I suggested [above](https://github.com/aspnet/Razor/pull/692/files#r53804803) could also avoid code duplication / reparsing.